File: java-streams/examples/java/src/main/java/streams/Imperative.java
package streams;

import java.util.ArrayList;
import java.util.List;

public class Imperative {

    List<Employee> findEmployees(List<Person> persons) {

        List employees = new ArrayList();
        for (Person person : persons) {
            if (person.isEmployed()) {
                Employee employee = new Employee(person.getFirstName() + " " + person.getLastName());
                employees.add(employee);
            }
        }
        return employees;
    }
}

File: java-streams/examples/kotlin/src/main/kotlin/streams/Imperative.kt
package streams

class Imperative {

    internal fun findEmployees(persons: List<Person>): List<Employee> {

        val employees = mutableListOf<Employee>()
        for ((_, firstName, lastName, isEmployed) in persons) {
            if (isEmployed) {
                val employee = Employee(firstName + " " + lastName)
                employees += employee
            }
        }
        return employees
    }
}
